### PR TITLE
[PAP-181] Add an OpenAlex scraper

### PR DESCRIPTION
**Pull request by notoraptor**
@breuleux Voici une PR pour PAP-181 !

### CODE REVIEW

File: paperoni/sources/scrapers/openalex.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+                    ),
+                    pages=None,
+                )
+                for location in release_locations
+            ],
+            topics=[
+                Topic(name=data_concept["display_name"])
+                for data_concept in data["concepts"]
+            ],
+            links=[Link(type=typ, link=ref) for typ, ref in data["ids"].items()]
```
Comment (breuleux) [inline paperoni/sources/scrapers/openalex.py:259]: It seems that the ids are all URLs, which is not consistent with what Paperoni does, e.g. `data[ids][doi] == 'https://doi.org/10.48550/arxiv.2111.09266'` but we want `10.48550/arxiv.2111.09266`, the part after the `/`. I think a decent general rule would be `ref.split("/", 3)[-1]` (we keep what is after the third slash).

The same goes for an author's `id` and `orcid` when present.

**Comment by breuleux** [comment]
It would be a good idea to use this:

https://docs.openalex.org/how-to-use-the-api/rate-limits-and-authentication#the-polite-pool

Maybe add a `mailto` field to `PaperoniConfig`, and use it if it's set.

**Comment by notoraptor** [comment]
> It would be a good idea to use this:
> 
> https://docs.openalex.org/how-to-use-the-api/rate-limits-and-authentication#the-polite-pool
> 
> Maybe add a `mailto` field to `PaperoniConfig`, and use it if it's set.

Done ! https://github.com/mila-iqia/paperoni/pull/61/commits/38912eb1fae52fc3a036c4fade4a9a3e3111d168

File: paperoni/sources/scrapers/openalex.py (hunk 1 above)
Comment (breuleux) [reply paperoni/sources/scrapers/openalex.py]: Yes, that's perfect!

**Pull request merged**